### PR TITLE
Periodic logger

### PR DESCRIPTION
**Pull request by chrisnojima**
This changes out store logging strategy. Instead of logging the store before/action every action we keep a ring buffer of the last 100 items.
We print out the state after every 50 changes
We have the ability to dump all 100 items on a log send (need rpc from core to make this work)
There is a debug flag to dump the objects immediately to the store (doesn't use the immutable transformer)

@keybase/react-hackers let's discuss

### CODE REVIEW

**Comment by chrisnojima** [comment]
- [ ] Need log send rpc from core: cc @maxtaco

File: shared/util/periodic-logger.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    this._size = size
+    this._name = name
+    this._logIncoming = logIncoming
+    this._logTransform = logTransform || ((args: Array<any>) => args)
+    this._logAfterXActions = logAfterXActions
+  }
+
+  _write (args: Array<any>) {
+    if (this._lastWrite === -1) {
+      this._lastWrite = 0
```
Comment (chromakode) [inline shared/util/periodic-logger.js]: Isn't this the same as `this._lastWrite++`?

File: shared/store/action-logger.js [hunk 2]
```diff
@@ -51,37 +15,17 @@ function makeActionToLog (action, oldState) {
...
-  const shouldRunLogStats = shouldRunStats(logStatFrequency)
+const logger = enableActionLogging ? setupLogger('actionLogger', 100, immediateStateLogging, immutableToJS, 50) : {log: () => {}}
 
+export const actionLogger = (store: any) => (next: any) => (action: any) => {
   const oldState = store.getState()
-
   const actionToLog = makeActionToLog(action, oldState)
+
   const log1 = [`Dispatching action: ${action.type}: `, forwardLogs ? JSON.stringify(actionToLog) : actionToLog]
+  console.log(log1)
```
Comment (chromakode) [inline shared/store/action-logger.js]: I'm a little confused by this. Shouldn't we be using `logger.log` here, and not the state below?

**Comment by chrisnojima** [comment]
We log actions immediately. They're cheap to log.

**Pull request merged**